app/lib/task/backend.dart

```diff
@@ -738,28 +739,33 @@ class TaskBackend {
   Future<BlobIndex?> _taskResultIndex(String package, String version) async =>
       await cache.taskResultIndex(package, version).get(() async {
         // Try runtimeVersions in order of age
-        var path = '$runtimeVersion/$package/$version/index.json';
-        List<int>? bytes;
         for (final rt in acceptedRuntimeVersions) {
```

Comment:
Hmm,

We could also avoid caching packageStatus per runtimeVersion, instead we could choose to only load the one for the current runtimeVersion.

If the version is not going to be analyzed by the current runtimeVersion, we should never load an analysis done by an old runtimeVersion, because that analysis will not eventually become consistent. That analysis is permanently stale.

So maybe we should just check the `packageState` of the current package, before we do:
`for (final rt in acceptedRuntimeVersions) {`

That'll probably shortcut must of the lookups, all of the rest of the lookups should eventually be able to find a result from the current runtimeVersion.